tsfile/src/main/java/org/apache/iotdb/tsfile/read/filter/operator/Like.java

```diff
@@ -51,7 +51,28 @@ public Like(String value, FilterType filterType) {
     this.value = value;
     this.filterType = filterType;
     try {
-      this.pattern = Pattern.compile(this.value);
+      String regexpStr = ".^$*+?{}[]|()";
+      StringBuilder patternStrBuild = new StringBuilder();
+      patternStrBuild.append("^");
+      for (int i = 0; i < value.length(); i++) {
+        String ch = String.valueOf(value.charAt(i));
+        if (regexpStr.contains(ch)) ch = "\\" + value.charAt(i);
+        if ((i == 0)
+            || (i > 0 && !"\\".equals(String.valueOf(value.charAt(i - 1))))
+            || ((i >= 4)
+                && "\\\\\\\\"
+                    .equals(
+                        patternStrBuild.substring(
+                            patternStrBuild.length() - 4, patternStrBuild.length())))) {
```

Comment:
Add more comments
```suggestion
        if ((i == 0)
            //  To avoid convert "\%"
            || (i > 0 && !"\\".equals(String.valueOf(value.charAt(i - 1))))
            // For case like "%\\\\%"
            || ((i >= 4)
                && "\\\\\\\\"
                    .equals(
                        patternStrBuild.substring(
                            patternStrBuild.length() - 4, patternStrBuild.length())))) {
```